pkg/registry/pod/etcd/etcd.go

```diff
@@ -56,17 +55,20 @@ func NewREST(h tools.EtcdHelper, factory pod.BoundPodFactory) (*REST, *BindingRE
 		},
 		EndpointName: "pods",
 
-		CreateStrategy: pod.Strategy,
+		Helper: h,
+	}
+	statusStore := *store
 
-		UpdateStrategy: pod.Strategy,
-		AfterUpdate:    bindings.AfterUpdate,
+	bindings := &podLifecycle{h}
+	store.CreateStrategy = pod.Strategy
+	store.UpdateStrategy = pod.Strategy
+	store.AfterUpdate = bindings.AfterUpdate
+	store.ReturnDeletedObject = true
+	store.AfterDelete = bindings.AfterDelete
 
-		ReturnDeletedObject: true,
-		AfterDelete:         bindings.AfterDelete,
+	statusStore.UpdateStrategy = pod.StatusStrategy
```

Comment:
Does it work? StatusStrategy does not implement some methods from RESTUpdateStrategy:
1. NamespaceScoped() bool
2. AllowCreateOnUpdate() bool